src/main/java/dp/xlsx/DatasetFormatter.java

```diff
@@ -46,31 +47,74 @@ void format(Sheet sheet, V4File file, Metadata datasetMetadata, CellStyle headin
                 .sorted()
                 .collect(Collectors.toList());
 
-
         final Collection<String> timeLabels = file.getOrderedTimeLabels();
 
-        final Map<String, Row> timeRows = new HashMap<>();
-        int columnOffset = 0;
+        sheet.setDefaultColumnWidth(timeLabels.iterator().next().length() + COLUMN_WIDTH_PADDING_CHARS);
+
         int rowOffset = 0;
 
-        rowOffset = addMetadataRows(sheet, rowOffset);
+        rowOffset = addMetadata(sheet, datasetMetadata, headingStyle, headingRightAlignStyle, rowOffset);
+        rowOffset = createHeaderRow(sheet, file, datasetMetadata, valueStyle, timeLabels, rowOffset);
 
-        // Start off by placing the time on all rows
-        int i = 0;
-        for (String timeLabel : timeLabels) {
-            Row row = sheet.createRow(i + rowOffset + 1);
+        for (SortedGroup group : sortedGroups) {
+
+            int columnOffset = 0;
+
+            // add dimensions label
+            Row row = sheet.createRow(rowOffset);
             Cell cell = row.createCell(columnOffset);
             cell.setCellStyle(valueStyle);
-            cell.setCellValue(timeLabel);
-            timeRows.put(timeLabel, row);
-            i++;
+            cell.setCellValue(group.getTitle());
+            columnOffset++;
+
+            for (String timeTitle : timeLabels) {
+
+                Cell obs = row.createCell(columnOffset);
+                setObservationCellValue(valueStyle, numberStyle, group, timeTitle, obs);
+                columnOffset++;
+            }
+
+            rowOffset++;
+        }
+
+        final int widestGroupTitle = sortedGroups.stream()
+                .mapToInt(g -> g.getTitleWidth())
+                .max()
+                .getAsInt();
+
+        sheet.setColumnWidth(0, widestGroupTitle * 256);
+    }
+
+    private void setObservationCellValue(CellStyle valueStyle, CellStyle numberStyle, SortedGroup group, String timeTitle, Cell obs) {
+        final String value = group.getObservation(timeTitle);
```

Comment:
if this happened before calling this function I think it would save passing group and timeTitle in? so this function could focus on just setting the value and styles on the cell.

Reply:
good point, this was my sloppy method extraction. I have made the change 👍